crowdfunding/projects/views.py

```diff
@@ -1,3 +1,102 @@
+from .serializers import ProjectSerializer, PledgeSerializer, ProjectDetailSerializer , PledgeDetailSerializer, CommentSerializer
+from .permissions import IsOwnerOrReadOnly, IsSupporterOrReadOnly, IsCommenterOrReadOnly
+
+from rest_framework.exceptions import NotFound
+from django.db import IntegrityError #unique = True handling
+
+from django_filters.rest_framework import DjangoFilterBackend
+
+#-- API-Root Config
+from rest_framework.decorators import api_view
+from rest_framework.reverse import reverse
+from users.models import CustomUser
+from users.serializers import CustomUserSerializer
+
+class ProjectList(generics.ListCreateAPIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+    filterset_fields = ['is_open', 'owner']
+
+    def perform_create(self, serializer):
+        serializer.save(owner=self.request.user)
+        # auto adds userid as owner
+
+    def post(self, request):
+        serializer = ProjectSerializer(data=request.data)
+        # serialize data for me
+        if serializer.is_valid():
+            try:
+                serializer.save(owner=request.user)
+                return Response(serializer.data, status=status.HTTP_201_CREATED)
+            except IntegrityError:
+                #responding to unique field
+                return Response({"error":"This Project title already exists. Please enter another."}, status=status.HTTP_400_BAD_REQUEST)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class ProjectDetail(generics.RetrieveUpdateDestroyAPIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly]
+    queryset = Project.objects.all()
+    serializer_class = ProjectDetailSerializer
+
+    def handle_exception(self, exc):
+        if isinstance(exc, Http404):
+            return Response({'data': 'Sorry, no tree-hugging here!'},
+            status=status.HTTP_404_NOT_FOUND)
+        return super(ProjectDetail, self).handle_exception(exc)
+
+class PledgeList(generics.ListCreateAPIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+    queryset = Pledge.objects.all()
+    serializer_class = PledgeSerializer
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = ('supporter', 'project')
+
+    def perform_create(self, serializer):
+        # added to remove the need to input a supporter {automates to logged in user}
+        serializer.save(supporter=self.request.user)
+
+class PledgeDetailView(generics.RetrieveUpdateDestroyAPIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly, IsSupporterOrReadOnly]
+    queryset = Pledge.objects.all()
+    serializer_class = PledgeDetailSerializer
+
+    def handle_exception(self, exc):
+        if isinstance(exc, Http404):
+            return Response({'data': 'Sorry, no tree-hugging here!'},
+            status=status.HTTP_404_NOT_FOUND)
```

Comment:
if you were going to do this for all of your classes, why not just put it as your default 404 message? 
you know the one i complained about earlier for not being json?